openeducat_activity/wizard/student_migrate_wizard.py

```diff
@@ -43,48 +57,79 @@ def _check_admission_register(self):
-                activity_vals = {
-                    'student_id': student.id,
-                    'type_id': act_type.id,
-                    'date': self.date,
-                    'description': 'Migration From' +
-                    record.course_from_id.name +
-                    ' to ' + record.course_to_id.name
-                }
+                if record.course_to_id:
+                    activity_vals = {
+                        'student_id': student.id,
+                        'type_id': act_type.id,
+                        'date': self.date,
+                        'description': 'Migration From' +
+                                       record.course_from_id.name +
+                                       ' to ' + record.course_to_id.name
+                    }
+                else:
+                    activity_vals = {
+                        'student_id': student.id,
+                        'type_id': act_type.id,
+                        'date': self.date,
+                        'description': 'Migration From' +
+                                       record.course_from_id.name +
+                                       ' to Completed Course'}
                 self.env['op.activity'].create(activity_vals)
-                student_course = self.env['op.student.course'].search(
-                    [('student_id', '=', student.id),
-                     ('course_id', '=', record.course_from_id.id)])
-                student_course.write({
-                    'course_id': record.course_to_id.id,
-                    'batch_id': record.batch_id.id})
-                reg_id = self.env['op.subject.registration'].create({
-                    'student_id': student.id,
-                    'batch_id': record.batch_id.id,
-                    'course_id': record.course_to_id.id,
-                    'min_unit_load': record.course_to_id.min_unit_load or 0.0,
-                    'max_unit_load': record.course_to_id.max_unit_load or 0.0,
-                    'state': 'draft',
-                })
-                reg_id.get_subjects()
-                if not record.optional_sub:
-                    reg_id.action_submitted()
-                    reg_id.action_approve()
+
+                if self.course_completed:
+                    for course_update in student.course_detail_ids:
+                        if course_update.course_id == self.course_from_id:
+                            course_update.state = 'finished'
+                else:
+                    for course_update in student.course_detail_ids:
+                        if course_update.course_id == self.course_from_id:
+                            course_update.state = 'finished'
+
+                            activity_vals = {
+                                'student_id': student.id,
+                                'type_id': act_type.id,
+                                'date': self.date,
+                                'description': 'Migration From' +
```

Comment:
User "  _(  ) ""for making the text translatable and don't use + to concat the string, use place holder and formatter.